src/combinatorics/antichain.lean

```diff
@@ -0,0 +1,80 @@
+/-
+Copyright (c) 2020 Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Bhavik Mehta, Alena Gusakov
+-/
+import data.finset
+import data.fintype.basic
+
+/-!
+# Antichains
+
+Basic definitions for finite sets which are useful for combinatorics.
+We define antichains, and a proposition asserting that a set is a set of r-sets.
+
+## Main definitions
+
+* `antichain` is a family of sets where no set is a subset of another.
+* `all_sized` is a proposition that
+
+-/
+
+open finset
+
+variable {α : Type*}
+variable {r : ℕ}
+
+/--
+A family of sets is an antichain if no set is a subset of another. For example,
+{{1}, {4,6,7}, {2,4,5,6}} is an antichain.
+-/
+def antichain (𝒜 : finset (finset α)) : Prop := ∀ A ∈ 𝒜, ∀ B ∈ 𝒜, A ≠ B → ¬(A ⊆ B)
```

Comment:
Does lean allow `⊈`?
```suggestion
def antichain (𝒜 : finset (finset α)) : Prop := ∀ A ∈ 𝒜, ∀ B ∈ 𝒜, A ≠ B → A ⊈ B
```

Reply:
I think even better would be to have `A ⊆ B → A = B`

Reply:
Or perhaps `A ≠ B → disjoint A B`?

Reply:
> It could be a good idea to generalise this to work for any partial order, not just the partial order on `finset a`

Do you think it would still be okay to have something like `def antichain (𝒜 : finset α) : Prop := ∀ A ∈ 𝒜, ∀ B ∈ 𝒜, A ≤ B → A = B`? Or should I try to make it so that `antichain` is defined on `(𝒜 : set α)`?